src/cosl/coordinated_workers/worker.py

```diff
@@ -406,6 +406,12 @@ def _add_readiness_check(self, new_layer: Layer):
 
     def _reconcile(self):
         """Run all unconditional logic."""
+        # There could be a race between the resource patch and pebble operations
+        # i.e., charm code proceeds beyond a can_connect guard, and then lightkube patches the statefulset
+        # and the workload is no longer available
+        if self.resources_patch and not self.resources_patch.is_ready():
+            logger.debug("Resource patching is not yet ready. Skipping config update.")
```

Comment:
```suggestion
            logger.debug("Resource patch not ready yet. Skipping reconciliation step since the pod will shortly be restarted.")
```

Reply:
I also think we should surface this in the status. something like "waiting for resource patch to apply". other charms do that too I think

Reply:
So, that should already happen. `line 277 inside _on_collect_status()`, The status would be WaitingStatus("waiting 
 for bla bla") or BlockedStatus("failed bla bla").

Reply:
I would change the debug message a bit and not mention "since the pod will shortly be restarted.", as it might have failed. that would be captured in the status message